### PR TITLE
MDN CSS: Put cleaned title in redirect field.

### PR DESCRIPTION
**Pull request by hchienjo**
Put the cleaned title in the redirect field so that it points to the right article.
 
Please not that I have not generated the output.txt after making this change.

cc @moollaza 

https://duck.co/ia/view/mdn_css

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@hchienjo this looks reversed. What I'd like to have is the **_original_** title, meaning the title **with** all non-alphanumeric characters (e.g. `:`, `@`, `()`) be the `title` for the Article (`'A'`) we create.

All other forms of that (i.e. `$title_clean`, `$title_clean function`, `$title function`, ...) should redirect to `title`

That way the final title displayed to the user will match the page title on MDN.

e.g "css dir" will redirect to ":dir()" and the Abstract displayed will have a title of **:dir()**

Does that make sense?

File: lib/fathead/mdn_css/parse.pl [hunk 1]
```diff
@@ -236,27 +236,25 @@ sub make_redirect {
...
-    my ($title) = @_;
+    my ( $title, $title_clean ) = @_;
     my @data;
     my $outputline;
     if ( $title =~ m/\(\)$/ ) {
         say "\nFound Function: $title";
         my $temp = $title;
         $temp =~ s/\(\)$/ function/;
         $outputline = join "\t",
-          ( $temp, 'R', $title, '', '', '', '', '', '', '', '', '', '' );
```
Comment (moollaza) [inline lib/fathead/mdn_css/parse.pl]: this line should stay 👍

File: lib/fathead/mdn_css/parse.pl [hunk 2]
```diff
@@ -236,27 +236,25 @@ sub make_redirect {
...
+          ( $temp, 'R', $title_clean, '', '', '', '', '', '', '', '', '', '' );
         push @data, $outputline;
         $temp = $title;
         $temp =~ s/\(\)$//;
         $outputline = join "\t",
-          ( $title, 'R', $temp, '', '', '', '', '', '', '', '', '', '' );
```
Comment (moollaza) [inline lib/fathead/mdn_css/parse.pl]: this should be flipped so `$temp` (e.g. `dir`) redirects to `$title` (e.g. `dir()`)

File: lib/fathead/mdn_css/parse.pl [hunk 3]
```diff
@@ -236,27 +236,25 @@ sub make_redirect {
...
+          ( $title, 'R', $title_clean, '', '', '', '', '', '', '', '', '', '' );
         push @data, $outputline;
     }
     elsif ( $title =~ qr/@/ ) {
-        my $temp = $title;
-        $temp =~ s/@//;
         $outputline = join "\t",
-          ( $title, 'R', $temp, '', '', '', '', '', '', '', '', '', '' );
```
Comment (moollaza) [inline lib/fathead/mdn_css/parse.pl]: `$temp` and `$title` should be reversed here

**Comment by hchienjo** [comment]
@moollaza Yes. Makes sense.

**Comment by moollaza** [comment]
@hchienjo thanks for this! It looks correct now.

Unfortunately our backend is having problems working with these article titles though. I've deployed it to Beta, but you'll notice some searches have stopped working.

I've made an issue for the team to fix this behaviour 👍

**Pull request merged**